__tests__/ResourceStateBuilder.test.js

```diff
@@ -35,23 +43,91 @@ describe('ResourceStateBuilder', () => {
+
+    const builder = new ResourceStateBuilder(dataset, '')
+    const result = await builder.buildState()
+    expect(result[0]).toEqual({
+      'resourceTemplate:bf2:Note': {
+        'http://www.w3.org/2000/01/rdf-schema#label': {
+          items: {
+            abc123: {
+              content: 'foo',
+              label: 'foo',
+              lang: 'en',
+            },
+          },
+        },
+      },
+    })
+  })
+
+  it('raises when 0 resource templates', async () => {
+    const resource = `<http://example/123> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://id.loc.gov/ontologies/bibframe/Note> .
+  <http://example/123> <http://www.w3.org/2000/01/rdf-schema#label> "bar"@en .`
+
+    const dataset = await rdfDatasetFromN3(resource)
+
+    const builder = new ResourceStateBuilder(dataset, 'http://example/123')
+    await expect(builder.buildState()).rejects.toMatch(/A single resource template must be included/)
+  })
+
+  it('raises when more than 1 resource templates', async () => {
+    const resource = `<http://example/123> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://id.loc.gov/ontologies/bibframe/Note> .
+    <http://example/123> <http://sinopia.io/vocabulary/hasResourceTemplate> "resourceTemplate:bf2:Note" .
+    <http://example/123> <http://sinopia.io/vocabulary/hasResourceTemplate> "resourceTemplate:bf2:Note2" .
+    <http://example/123> <http://www.w3.org/2000/01/rdf-schema#label> "bar"@en .`
+
+    const dataset = await rdfDatasetFromN3(resource)
+
+    const builder = new ResourceStateBuilder(dataset, 'http://example/123')
+    await expect(builder.buildState()).rejects.toMatch(/A single resource template must be included/)
+  })
+
+  it('raises when error loading resource templates', async () => {
+    const resource = `<http://example/123> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://id.loc.gov/ontologies/bibframe/Note> .
+    <http://example/123> <http://sinopia.io/vocabulary/hasResourceTemplate> "resourceTemplate:bf2:Note" .
+    <http://example/123> <http://www.w3.org/2000/01/rdf-schema#label> "bar"@en .`
+
+    sinopiaServer.getResourceTemplate.mockRejectedValue(new Error('404'))
+
+    const dataset = await rdfDatasetFromN3(resource)
+
+    const builder = new ResourceStateBuilder(dataset, 'http://example/123')
+    await expect(builder.buildState()).rejects.toMatch(/Error getting resourceTemplate:bf2:Note/)
   })
 
+
   it('builds the state for URI resource properties', async () => {
     const resource = `<http://example/123> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://id.loc.gov/ontologies/bibframe/Note> .
   <http://example/123> <http://sinopia.io/vocabulary/hasResourceTemplate> "resourceTemplate:bf2:Note" .
-  <http://example/123> <http://rdaregistry.info/Elements/i/P40021> <http://id.loc.gov/vocabulary/organizations/wauar> .`
+  <http://example/123> <http://www.w3.org/2000/01/rdf-schema#label> <http://id.loc.gov/vocabulary/organizations/wauar> .`
```

Comment:
this triple doesn't have a valid range: https://www.w3.org/TR/rdf-schema/#ch_label

Reply:
Added a note to test about this.

Reply:
why not just change it to a literal?

Reply:
I guess I'm having a hard time understanding the use case that this test is representing and by making the data more realistic it would help demonstrate exactly what behavior we're looking for.

Reply:
Because the test case is for loading a URI. However, we don't have a test resource template for URIs (and I'm not even sure what would be a realistic template to create).